src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/MethodCodeNode.cs

```diff
@@ -99,6 +99,26 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
                     dependencies.Add(new DependencyListEntry(factory.MethodEntrypoint(invokeStub), "Reflection invoke"));
             }
 
+            if (_method.HasInstantiation)
+            {
+                if (factory.MetadataManager.ExactMethodInstantiations.AddExactMethodInstantiationEntry(factory, _method))
```

Comment:
Computing dependencies shouldn't have this sort of side effects - we might want an abstract node (abstract in the sense that it doesn't derive from ObjectNode, but from DependencyNodeCore) that will do the Add in it's OnMarked method. Same way we do the on-demand virtual methods.

I need to familiarize myself with stuff that already got merged to provide more feedback, but I'm at the airport...